packages/sheet2code-ui/webpack.config.js

```diff
@@ -46,15 +46,16 @@ const webpackConfig = {
     main: path.resolve(__dirname, './src/bundle-entry.js'),
   },
   output: {
-    filename: isProduction
-      ? '[name].[chunkhash].bundle.js'
-      : '[name].[hash].bundle.js',
+    filename: '[name].[hash].bundle.js',
     path: path.resolve(__dirname, './dist/'),
+    chunkFilename: '[name].[chunkhash].chunk.js',
+    library: pkg.name,
+    libraryTarget: 'umd',
   },
   optimization: {
     minimize: isProduction,
     splitChunks: {
-      chunks: 'all',
+      chunks: 'initial',
```

Comment:
Just to confirm my understanding of `optimization.splitChunks.chunks`:

`'async'` -> only the modules imported inside `import()` will be abstracted out
`'initial'` -> only the modules imported by regular `import ... from '...'` will be abstracted out
`'all'` -> all modules imported by both above ways will be abstracted out

Is my understanding right? I'm not very sure when reading the [document](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkschunks)

Reply:
I am not 100% sure since I haven't tested it (in webpack v4) so far.
But I think you are right about the document.

When we set `chunks: 'async'`, the following `cacheGroups` config would only apply on those `async`(`import()` or `require.ensure`) chunks, not including `initial` chunks (`import from` or `require()`)

Reply:
Just tested it. And just like what we say.
`async` -> cacheGroups applies on `async` chunks
`initial` -> cacheGroups applies on non-`async` chunks
`all` -> cacheGroups applies on `async` and non-`async`

Reply:
The split chunks look very different if we set `chunks` to different value.
If we want to use dynamic import in the future, `cacheGroups` settings  needs to be revised case by case.